ipn/backend.go

```diff
@@ -187,6 +187,14 @@ type Options struct {
 	//    an initial overwrite of backend state with Prefs.
 	StateKey StateKey
 	Prefs    *Prefs
+	// UpdatePrefs, if provided, overrides the default Prefs stored in
```

Comment:
Document the relation to Options.Prefs; is Prefs still used? (it currently is)  Say exactly how they're supposed to interact.

I really don't like this, though. We've been trying to chip away at Start's specialness (compare all the ipn.Options in 1.4: https://github.com/tailscale/tailscale/blob/v1.4.0/ipn/backend.go#L101) and this goes in the other direction, adding more specialness to Start.

Reply:
As discussed, this is a mess and I really don't think we want the separate Prefs and UpdatePrefs fields long term. Added a longer comment describing possible fixes.